### PR TITLE
Nguyenquyhy/17 custom linear gauges

### PR DESCRIPTION
**Pull request by rmroc451**
well, here is my clean up and consolidation effort after the last one 👍 

Sorry for the delay, life hit hard!

here is an example profile (as well as my personal profile) and all my icons if you want to include that on the release.  No biggie if you don't.
[17-Examples and icons you can distribute.zip](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/5577145/17-Examples.and.icons.you.can.distribute.zip)

If you have more concerns, let me know.

image

### CODE REVIEW

**Comment by nanostra** [comment]
Hello,

When I install your sample "17-Examples and icons you can distribute.zip"... 

- I install your folder inside C:\Users\frede\AppData\Roaming\Elgato\StreamDeck\Plugins... 
- I don't have your Linear gauge when I import "Custom Linear Gauge Example"... but standard gauge... as the original project..

May be I have to install your release... but in the user guide, the link Redirect to the Original Project... 

Could you help me ?

Best regards.
Fred

Snag_10b701c

**Comment by rmroc451** [comment]
Hi! You'd have to download and compile the source for these to work.

They haven't been merged in yet, so they won't work until a release is given so you can install it.

**Comment by nanostra** [comment]
Then, it's normal, if there is no binaries :-)
Thank you for your work... I will wait for the next release... great work !

Best regards,
Fred

File: FlightStreamDeck.Logics/Actions/GenericGaugeAction.cs [hunk 1]
```diff
@@ -16,10 +16,42 @@ public class GenericGaugeSettings
...
+                MinValue == 0 &&
+                MaxValue == 0 &&
+                ChartThicknessValue == 0 &&
+                ChartChevronSizeValue == 0 && !DisplayHorizontalValue;
+        }
     }
 
     [StreamDeckAction("tech.flighttracker.streamdeck.generic.gauge")]
-    public class GenericGaugeAction : StreamDeckAction<GenericGaugeSettings>
+    public class GenericGaugeAction : StreamDeckAction<CustomGaugeSettings>
```
Comment (nguyenquyhy) [inline FlightStreamDeck.Logics/Actions/GenericGaugeAction.cs:54]: It's a bit nitpicking but can you help to change this back to GenericGaugeSettings? I would really want to keep the convention of GenericGaugeAction ~ GenericGaugeSettings to not confuse myself.

I think GenericGaugeSettings is not used anywhere now other than a base class for CustomGaugeSettings, so I would prefer merging CustomGaugeSettings's property into GenericGaugeSettings.

Comment (rmroc451) [reply FlightStreamDeck.Logics/Actions/GenericGaugeAction.cs:54]: On it.

**Pull request merged**